dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/user/UserController.java

```diff
@@ -445,66 +450,39 @@ public void disableUser( @PathVariable("uid") String uid )
-        {
-            throw new UpdateAccessDeniedException( "You don't have the proper permissions to update this user." );
-        }
-
-        // force initialization of all authorities of current user in order to prevent cases where user must be reloaded later
-        // (in case it gets detached)
-        if ( currentUser != null )
-        {
-            currentUser.getUserCredentials().getAllAuthorities();
-        }
-
         User parsed = renderService.fromXml( request.getInputStream(), getEntityClass() );
-        parsed.setUid( pvUid );
-
-        parsed = mergeLastLoginAttribute( users.get( 0 ), parsed );
 
-        if ( !userService.canAddOrUpdateUser( IdentifiableObjectUtils.getUids( parsed.getGroups() ), currentUser )
-            || !currentUser.getUserCredentials().canModifyUser( users.get( 0 ).getUserCredentials() ) )
-        {
-            throw new WebMessageException( WebMessageUtils.conflict( "You must have permissions to create user, or ability to manage at least one user group for the user." ) );
-        }
-
-
-        MetadataImportParams params = importService.getParamsFromMap( contextService.getParameterValuesMap() );
-        params.setImportReportMode( ImportReportMode.FULL );
-        params.setImportStrategy( ImportStrategy.UPDATE );
-        params.addObject( parsed );
-
-        ImportReport importReport = importService.importMetadata( params );
-
-        updateUserGroups( importReport, pvUid, parsed, currentUser );
+        ImportReport importReport = updateUser( pvUid, parsed );
 
         response.setContentType( "application/xml" );
-
         renderService.toXml( response.getOutputStream(), importReport );
     }
 
     @Override
     @RequestMapping( value = "/{uid}", method = RequestMethod.PUT, consumes = "application/json" )
     public void putJsonObject( @PathVariable( "uid" ) String pvUid, HttpServletRequest request, HttpServletResponse response ) throws Exception
     {
-        List<User> users = getEntity( pvUid, NO_WEB_OPTIONS );
+        User parsed = renderService.fromJson( request.getInputStream(), getEntityClass() );
+
+        ImportReport importReport = updateUser( pvUid, parsed );
+
+        response.setContentType( "application/json" );
+        renderService.toJson( response.getOutputStream(), importReport );
+    }
+
+    protected ImportReport updateUser( String userUid, User parsedUserObject )
+        throws WebMessageException
+    {
+        List<User> users = getEntity( userUid, NO_WEB_OPTIONS );
 
         if ( users.isEmpty() )
         {
-            throw new WebMessageException( WebMessageUtils.conflict( getEntityName() + " does not exist: " + pvUid ) );
+            throw new WebMessageException(
+                WebMessageUtils.conflict( getEntityName() + " does not exist: " + userUid ) );
         }
 
         User currentUser = currentUserService.getCurrentUser();
-
+        // TODO: Can we disallow currentUser == NULL ?
```

Comment:
If anyone have any suggestions or comments that would be nice, I would very much like this to not be allowed i.e. currentUser == NULL > cast exception